webapp/tests/karma/ts/modules/analytics/analytics-route-guard.provider.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { Selectors } from '@mm-selectors/index';
+
+describe('AnalyticsRouteGuard provider', () => {
+  let provider: AnalyticsRouteGuardProvider;
+  let store;
+  let router;
+
+  beforeEach(() => {
+    store = { select: sinon.stub() };
+    router = { createUrlTree: sinon.stub() };
+    provider = new AnalyticsRouteGuardProvider(store, router);
+  });
+
+  afterEach(() => sinon.restore());
+
+  it('should return module path if single module is present', async(() => {
+    const route = new ActivatedRouteSnapshot();
+    route.data = { tab: 'analytics' };
+    const subject = new Subject();
+    store.select.returns(subject);
+    router.createUrlTree.returnsArg(0);
+
+    provider
+      .canActivateChild(route)
+      .subscribe(result => {
+        expect(result).to.deep.equal([ '/', 'analytics', 'targets' ]);
+        expect(router.createUrlTree.callCount).to.equal(1);
+        expect(router.createUrlTree.args[0]).to.deep.equal([[ '/', 'analytics', 'targets' ]]);
+      });
+
+    subject.next([ { id: 'targets', route: ['/', 'analytics', 'targets'] } ]);
+
+    expect(store.select.callCount).to.equal(1);
+    expect(store.select.args[0]).to.deep.equal([Selectors.getAnalyticsModules]);
+  }));
+
+  it('should return true and not module path if multiple modules are present', async(() => {
+    const route = new ActivatedRouteSnapshot();
+    route.data = { tab: 'analytics' };
+    const subject = new Subject();
+    store.select.returns(subject);
+    router.createUrlTree.returnsArg(0);
+
+    provider
+      .canActivateChild(route)
+      .subscribe(result => {
+        expect(result).to.be.true;
+        expect(router.createUrlTree.callCount).to.equal(0);
+      });
+
+    subject.next([
+      { id: 'target-aggregates', route: ['/', 'analytics', 'target-aggregates'] },
+      { id: 'targets', route: ['/', 'analytics', 'targets'] }
+    ]);
+
+    expect(store.select.callCount).to.equal(1);
+    expect(store.select.args[0]).to.deep.equal([Selectors.getAnalyticsModules]);
+  }));
+
+  it('should return true and not module path if it is not analytics tab', async(() => {
+    const route = new ActivatedRouteSnapshot();
+    route.data = { tab: 'messages' };
+    const subject = new Subject();
+    store.select.returns(subject);
+    router.createUrlTree.returnsArg(0);
+
+    provider
+      .canActivateChild(route)
+      .subscribe(result => {
+        expect(result).to.be.true;
+        expect(router.createUrlTree.callCount).to.equal(0);
+      });
+
+    subject.next([ { id: 'targets', route: ['/', 'analytics', 'targets'] } ]);
+
+    expect(store.select.callCount).to.equal(1);
+    expect(store.select.args[0]).to.deep.equal([Selectors.getAnalyticsModules]);
+  }));
+
+  it('should not emit if analytics modules are falsy', fakeAsync(() => {
+    const route = new ActivatedRouteSnapshot();
+    route.data = { tab: 'messages' };
+    const subject = new Subject();
```

Comment:
Very clean tests, I'm learning a lot about testing angular's routing and store mechanics :ok_hand: